### PR TITLE
py-sphinx: add v5.2.0

### PR DESCRIPTION
**Pull request by adamjstewart**
Successfully installs on macOS 12.6 (arm64) with Python 3.9.13 and Apple Clang 14.0.0.

https://www.sphinx-doc.org/en/master/changes.html

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
LGTM. Confirmed all three new sha256.

**Comment by tldahlgren** [comment]
Will defer merged until @takluyver has a chance to comment.

File: var/spack/repos/builtin/packages/py-flit-core/package.py [hunk 1]
```diff
@@ -23,10 +24,13 @@ class PyFlitCore(PythonPackage):
     version("3.0.0", sha256="a465052057e2d6d957e6850e9915245adedfc4fd0dd5737d0791bf3132417c2d")
     version("2.3.0", sha256="a50bcd8bf5785e3a7d95434244f30ba693e794c5204ac1ee908fc07c4acdbf80")
 
-    # Dependencies listed in flit_core/build_thyself.py
+    # pyproject.toml
+    depends_on("python@3.7:", when="@3.7:", type=("build", "run"))
```
Comment (takluyver) [inline var/spack/repos/builtin/packages/py-flit-core/package.py]: `flit_core` should still work on Python 3.6 - we have CI that aims to test this. If you've found that it doesn't, please let me know.

I try to be fairly conservative with `flit_core`, while sometimes `flit` (the CLI) requires a newer version of Python.

**Comment by takluyver** [comment]
I'm guessing I was tagged for the flit-core part, so I haven't looked at the other changes in any detail. Other than my one comment above, this LGTM.

File: var/spack/repos/builtin/packages/py-flit-core/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-flit-core/package.py]: Good catch, not sure where I got that from

**Review by tldahlgren — APPROVED** [review]

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Pull request merged**